lib/linux_admin/hosts.rb

```diff
@@ -36,6 +38,20 @@ def update_entry(address, hostname, comment = nil)
       end
     end
 
+    def hostname=(name)
```

Comment:
I would say that the safest way to go would be to do as little as possible with each call.  I don't really want to assume what people will want to do with these tools.  I think it will be more clear from the calling code to do something like this https://github.com/ManageIQ/manageiq/pull/4558/files#diff-16bd2bb3c0abfe4c0381896614e7b987R238